client/asset/dcr/rpcwallet.go

```diff
@@ -0,0 +1,526 @@
+	chainjson "github.com/decred/dcrd/rpc/jsonrpc/types/v3"
+	"github.com/decred/dcrd/rpcclient/v7"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+)
+
+// Implements Wallet functionality using rpc connections.
+type rpcWallet struct {
+	// 64-bit atomic variables first. See
+	// https://golang.org/pkg/sync/atomic/#pkg-note-BUG
+	tipAtConnect int64
+
+	chainParams *chaincfg.Params
+	log         dex.Logger
+
+	client *rpcclient.Client
+	node   rpcClient
+}
+
+var _ Wallet = (*rpcWallet)(nil)
+
+// rpcClient is an rpcclient.Client, or a stub for testing.
+type rpcClient interface {
+	EstimateSmartFee(ctx context.Context, confirmations int64, mode chainjson.EstimateSmartFeeMode) (*chainjson.EstimateSmartFeeResult, error)
+	GetBlockChainInfo(ctx context.Context) (*chainjson.GetBlockChainInfoResult, error)
+	SendRawTransaction(ctx context.Context, tx *wire.MsgTx, allowHighFees bool) (*chainhash.Hash, error)
+	GetTxOut(ctx context.Context, txHash *chainhash.Hash, index uint32, tree int8, mempool bool) (*chainjson.GetTxOutResult, error)
+	GetBalanceMinConf(ctx context.Context, account string, minConfirms int) (*walletjson.GetBalanceResult, error)
+	GetBestBlock(ctx context.Context) (*chainhash.Hash, int64, error)
+	GetBlockHash(ctx context.Context, blockHeight int64) (*chainhash.Hash, error)
+	GetBlockVerbose(ctx context.Context, blockHash *chainhash.Hash, verboseTx bool) (*chainjson.GetBlockVerboseResult, error)
+	GetBlockHeaderVerbose(ctx context.Context, blockHash *chainhash.Hash) (*chainjson.GetBlockHeaderVerboseResult, error)
+	GetRawMempool(ctx context.Context, txType chainjson.GetRawMempoolTxTypeCmd) ([]*chainhash.Hash, error)
+	GetRawTransactionVerbose(ctx context.Context, txHash *chainhash.Hash) (*chainjson.TxRawResult, error)
+	LockUnspent(ctx context.Context, unlock bool, ops []*wire.OutPoint) error
+	GetRawChangeAddress(ctx context.Context, account string, net stdaddr.AddressParams) (stdaddr.Address, error)
+	GetNewAddressGapPolicy(ctx context.Context, account string, gap dcrwallet.GapPolicy) (stdaddr.Address, error)
+	DumpPrivKey(ctx context.Context, address stdaddr.Address) (*dcrutil.WIF, error)
+	GetTransaction(ctx context.Context, txHash *chainhash.Hash) (*walletjson.GetTransactionResult, error) // Should return asset.CoinNotFoundError if tx is not found.
+	WalletLock(ctx context.Context) error
+	WalletPassphrase(ctx context.Context, passphrase string, timeoutSecs int64) error
+	AccountUnlocked(ctx context.Context, account string) (*walletjson.AccountUnlockedResult, error)
+	LockAccount(ctx context.Context, account string) error
+	UnlockAccount(ctx context.Context, account, passphrase string) error
+	Disconnected() bool
+	RawRequest(ctx context.Context, method string, params []json.RawMessage) (json.RawMessage, error)
+	WalletInfo(ctx context.Context) (*walletjson.WalletInfoResult, error)
+	ValidateAddress(ctx context.Context, address stdaddr.Address) (*walletjson.ValidateAddressWalletResult, error)
```

Comment:
There is another ValidateAddress in `dcr.go` still. Should it be removed?

Reply:
Ah, moved. Irrc, the `ValidateAddress` in dcr.go is the one to be used, it overrides the one here. But I'll take another look to be sure nothing has gone wrong there. Thanks for pointing this out.